Jenkinsfile

```diff
@@ -84,6 +84,7 @@ timestamps {
                 file(credentialsId:'ROBOTEST_LOG_GOOGLE_APPLICATION_CREDENTIALS', variable: 'GOOGLE_APPLICATION_CREDENTIALS'),
                 file(credentialsId:'OPS_SSH_KEY', variable: 'SSH_KEY'),
                 file(credentialsId:'OPS_SSH_PUB', variable: 'SSH_PUB'),
+                string(credentialsId:'GET_GRAVITATIONAL_IO_APIKEY', variable: 'BUILD_APIKEY'),
```

Comment:
We shouldn't need this to fetch images, as far as I know. `tele pull` will download the correct artifacts when it isn't logged in to get.gravitational.io -- so long as it isn't logged in elsewhere either.

We should test both authenticated & unauthenticated fetches, but if we only do one, unauthenticated are more important to customers in my opinion.  Especially for the new user experience.

Reply:
Do we always logout? Since if there is login state (from an unrelated job), this will still use it (this is what's been happening here).

Reply:
We did briefly, but that caused issues for publishing.

I think login is the way to go for now -- it is consistent with the old behavior.

I'm working on a patch that containerizes more of the robotest workflow such that `~jenkins` state cannot affect fetching robotest artifacts (and vice versa), but that is probably a day or two out -- it is a more substantial refactoring

Reply:
I have nothing against using login but it seems that publishing is a very focused task which is not necessary for PR jobs - I think explicit token configuration is as good as logging in in a per-task environment, no?

Reply:
> I have nothing against using login but it seems that publishing is a very focused task which is not necessary for PR jobs

Publishing & PR shouldn't be related, but are due to sharing login state in `~jenkins`.  See https://jenkins.gravitational.io/view/Publishing/job/Telekube-Publish-5.2/301/console for an example failure

> I think explicit token configuration is as good as logging in in a per-task environment, no?

My goal for test purposes would be to fetch artifacts while not logged in.  I've yet to find a good way to use unauthenticated pulls in a way that is non

On a more holistic note, I want to isolate because sharing state in `~/` has proven to be prone to races and a large pain point.  This stolon/pithos CI breaking robotest topic is the 3rd notable disruption stemming from "spooky action at a distance" when unrelated parts of our CI/release infra breaking each other due to sharing state in `~/`.  See https://github.com/gravitational/gravity/pull/1941 and https://github.com/gravitational/gravity/pull/177 for the first two.

While I *think* I understand enough to not make mistakes regarding this shared state after having been burned by it twice, will the next employee who touches our release infra understand?  Why leave this edge around for them to trip on?

Reply:
I think we're not following each other. I'm not saying we should use shared state and using hub/token does not create shared (login) state - it is a way to use the login information at that particular point where it is applied - isn't this what we're after?

Reply:
> I'm not saying we should use shared state and using hub/token does not create shared (login) state - it is a way to use the login information at that particular point where it is applied - isn't this what we're after?

A couple issues with the hub/token approach:

1) How do we apply this advice to 6.1 and 5.5 where the hub/token flags do not exist?  Those branches are subject to the same shared state modes of failure. 

2) How do we test unauthenticated fetch with the hub and token flags?  Related: How could open source users run any of these targets?

To be clear: I'm not asking you to change anything here in this PR.  We need an immediate mitigation and your PR provides this in the best way!  This discussion is about a long term plan.

Reply:
The answer to all the above could be "product fixes & backports to all active releases".  But per our earlier discussion about "do unnecessary feature backports" I'm inclined to look at non-product fixes to our build architecture.  Lets fix the product in master, but use non-product solutions (could be sandboxing pushes/pulls in a container, could be something else) to fix the issue in older branches.  This reduces surface area for product regressions, especially since this wouldn't be work a customer cares about -- it is purely internal work.